### DISCUSSION TITLE
Oracle Database connection no results

### DISCUSSION THREAD

**Discussion by parkinsonr** [nextgenhealthcare/connect#6052] [Q&A]
I'm attempting to connect to an oracle database and execute a select query. I believe I have it all configured correctly using the proper TNS info. Once deployed I can see the channel status 'Polling' for a couple minutes then 'Reading' for a few more minutes in which I would expect to see results, however it then goes back to 'Idle' status with no results. No errors showing in the server log either.

Running Mirth version 4.4.1.
Oracle version is 19.3.0.0.0

In the Oracle logs we're seeing the following error:

2024-01-05T15:59:05.375642-05:00
05-JAN-2024 15:59:05 * (ADDRESS=(PROTOCOL=tcp)(HOST=xx.xx.xx.x)(PORT=55560)) * <unknown connect data> * 12537
TNS-12537: TNS:connection closed
TNS-12560: TNS:protocol adapter error
  TNS-00507: Connection closed
   Linux Error: 115: Operation now in progress

Here's the connection Info I have setup on the channel using javascript

`var dbConn;

try {
	dbConn = DatabaseConnectionFactory.createDatabaseConnection('oracle.jdbc.driver.OracleDriver','jdbc:oracle:thin:@(DESCRIPTION = (ADDRESS_LIST = (ADDRESS = (PROTOCOL = TCP)(HOST = 10.60.9.39)(PORT = 30002)) (ADDRESS = (PROTOCOL = TCP)(HOST = 10.60.9.39)(PORT = 30002))) (CONNECT_DATA = (SID = ONCOREP) (SERVER = DEDICATED)))','username','pass');
	result = dbConn.executeCachedQuery("SELECT * FROM oncore.sv_pcl_status")
	return result;
} finally {
	if (dbConn) { 
		dbConn.close();
	}
}`

![image](https://github.com/nextgenhealthcare/connect/assets/57762761/99b767f1-9c14-4748-990b-7ec55e9ef288)

**Comment by ab-mg-23** [comment]
You have a duplicated `(ADDRESS)` in the connection description. Remove the duplicate and see if you still don't get results.

**Reply by parkinsonr** [reply]
Thanks for the suggestion. I removed the duplicate and it's still getting the same problem.

**Comment by Alexandraa72** [comment]
The issue can be resolved by setting the value of TRANSPORT_CONNECT_TIMEOUT to milliseconds instead of seconds. That is instead of
TRANSPORT_CONNECT_TIMEOUT=3
use
TRANSPORT_CONNECT_TIMEOUT=3000

 Example:

 

jdbc:oracle:thin:@(DESCRIPTION = (CONNECT_TIMEOUT = 5) (TRANSPORT_CONNECT_TIMEOUT = 3000) (RETRY_COUNT = 3) (RETRY_COUNT=20)(FAILOVER=ON) (ADDRESS_LIST =(ADDRESS=(PROTOCOL=tcp) (HOST=<HOSTNAME)(PORT=<PORT>)) (CONNECT_DATA=(SERVICE_NAME=<service-name>)))
